src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -373,6 +375,11 @@ def column_states(self, value: ColumnStateDict):
         """
         self._column_states = deepcopy(value)
 
+    @abstractmethod
+    def referred_cte_tables(self) -> Set[str]:
```

Comment:
```suggestion
    def reference_ctes(self) -> Set[str]:
```
I guess this name is more accurate?